darwin.m

```diff
@@ -0,0 +1,67 @@
+// +build darwin
+
+#import <Foundation/NSArray.h>
```

Comment:
elevate has a very simple API, does exactly what you need and uses the existing program cocoasudo so you don't have to write all this Objective C code yourself.  Are you confusing elevate and elevated?